internal/parser/types.go

```diff
@@ -25,5 +25,7 @@ type (
 		Comments []string
 	}
 
-	Type string
+	Type struct {
+		Name string
```

Comment:
Why this change ?
- To be future proof when we add more attributes into Type function (and no breaking change)
- Because we could not use string helpers in template in a "Type" type (eg. `.Type | upper` was broken because .Type is not of type string)